tasks/post-to-cmr/index.js

```diff
@@ -278,8 +287,9 @@ async function updateCmrFileAccessURls(cmrFiles, granulesObject) {
     cmrFile.metadataObject.Granule.OnlineAccessURLs.OnlineAccessURL = urls;
     const builder = new xml2js.Builder();
     const xml = builder.buildObject(cmrFile);
-    const [, Bucket, Key] = cmrFile.filename.match(/^s3:\/\/(.+?)\/(.+)$/);
-    await promiseS3Upload({ Bucket, Key, Body: xml });
+    const updatedCmrFile = granule.files.find((f) => f.filename.match(/.*\.cmr\.xml$/));
+
+    await promiseS3Upload({ Bucket: updatedCmrFile.bucket, Key: updatedCmrFile.name, Body: xml });
```

Comment:
Instead of `updatedCmrFile.name` this should probably be `updatedCmrFile.filepath`, as it has the updated keypath if a url_path template was applied to this file, otherwise falls back to whatever is in `.name`.